### ISSUE TITLE
💩 Hide shitcoins

### ISSUE THREAD

**Issue by ricburton** [balance-io/balance-manager#242]
I have 2 assets and 2 shitcoins. These should be hidden.
<img width="1009" alt="screen shot 2018-05-23 at 17 39 49" src="https://user-images.githubusercontent.com/438160/40452594-600c6986-5eb0-11e8-934b-893c53f2ea25.png">

**Comment by pedrouid** [comment]
This was a fairly recent change that me and Christian decided to unhide if the account had less than 5 tokens.

What we should probably do instead is enable the user to choose which tokens to hide? Or revert the change to hide according to Market Value regardless of total amount of tokens?

**Comment by amhed** [comment]
We could have a black list of the better known ones. And then give the user the option to remove them locally if they wanted to

**Comment by benjiqq** [comment]
do there exist standards/registers for whitelists? since symbol names have to be third party sources anyways, it makes sense that those sources are effective whitelists. the more notable attack is registering a name "ABC" for a token which has the name but did not register it yet. ZRX is well known, but there is no mechanism globally which matches ZRX to the token address without trusting a third party (etherscan, coinmarketcap, exchange, ...).

**Comment by pedrouid** [comment]
@benjyz Yes, that's true this is a huge issue, currently we are managing assets in Balance Manager (tokens and Ethereum) by their symbol, it's the simplest away to do it and most common as well.
However because we enable the user to view any token without shortlisting them, we have had issues specially with conflicting prices from cryptocompare with tokens with the same symbol/ticker.

@amhed  I don't agree that we should have a blacklist or whitelist because we want to be as decentralised and permissionless as possible. It's up to the users to manage whatever tokens they desire.

PROPOSAL: What we REALLY need is to create Token Curated Registry, or a more pragmatic approach, a repository with matching token symbols/names to their contract addresses. This would solve two majors issues: 
1. conflitcing prices with tokens of the same symbol/ticker
2. tokens trying to mimic or impersonate popular tokens (like @benjyz was mentioning)

**Comment by amhed** [comment]
Regarding the repo, I thought we already had that here: 
https://github.com/balance-io/tokens/

I can take a swing at allowing the customer to show/hide tokens. How about: 
- Adding an ellipsis icon on each row for the "assets" tab. Context menu has the hide option.
- Adding an "options" icon on top of the table and allow: 
  - Hide all assets with value < x
  - Hide all dust (asset qty < 0.0000x)
  - Hide a list of coins by name or by contract address (blacklist that stored only on customer's local storage)

**Comment by pedrouid** [comment]
That repo simply has icons by contract address. It doesn't link names to contract addresses.

Feel free to take a swing at that though

**Comment by amhed** [comment]
Here's how I plan on tackling: 

- Each row would show an additional icon on hover if you chose to hide the token:
![per-row](https://user-images.githubusercontent.com/579291/40581965-7bef955a-612c-11e8-8150-126ec4157dce.gif)

- The bottom link would change from "Hide x tokens with low/no market value" to "Hide/Show x Hidden tokens"

- We would hide the ones that have no/low market value like we do right now by default, and the user decided to add or remove to this list. That would get persisted in local storage

- Instead of showing the hidden icons on the top, show them after the link and allow adding back:
![shitcoin hiding proposal](https://user-images.githubusercontent.com/579291/40581972-d1fd13fa-612c-11e8-98b1-6900a3d7558e.png)

**Comment by amhed** [comment]
BTW, I got the SVGs for the icons from font-awesome. Is there any specific source we're getting our assets from? or are they being drawn in-house?

**Comment by benjiqq** [comment]
@pedrouid I can do some work on the curated registry and created a new issue https://github.com/balance-io/balance-manager/issues/248

**Comment by amhed** [comment]
Hi guys, I know you're all working on a bunch of stuff at the same time. I have 2 PRs out, looking for some feedback on this one and for a possible merge on the spanish translation one before picking up on something else 🚀

**Comment by ricburton** [comment]
Hi Amhed,

Thanks for your time! Pedro has been in Argentina for EthBA. We will get to this ASAP.

Richard

Sent via Superhuman ( https://sprh.mn/?vip=richard@balance.io )
